### ISSUE TITLE
tf.nn.ctc_beam_search_decoder is very slow on tf >=1.4

### ISSUE THREAD

**Issue by tianq01** [tensorflow/tensorflow#26200]
<em>Please make sure that this is a bug. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): yes 
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): ubuntu 16.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): tried 1.4, 1.5, 1.8, 1.9, 1.12
- Python version: 3.5
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version: CUDA8 for 1.4, CUDA9 for >=1.5
- GPU model and memory:P100

You can collect some of this information using our environment capture [script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh)
You can also obtain the TensorFlow version with
python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"

**Describe the current behavior**

Our text recognizer is based on CRNN model  and CTC. it was developed based on tf 1.3 and performs well.
recently we tried to upgrade the tensorflow to tf 1.9 and find the recognizer inference time increases a lot.
we identified the slow down is caused by tf.nn.ctc_beam_search_decoder op using this way:

for this line of code:
decoded, prob = tf.nn.ctc_beam_search_decoder(logits, seq_length)

running time for session.run(logits) is fine, however session.run(decoded) is very slow

we verified the problem occurrs on tf versions 1.12, 1.9, 1.8, 1.5, 1.4
we also verified the model ckpt trained with tf 1.9 is running inference fast with tf 1.3. similarly, model ckpt trained with tf 1.4/1.5 also running fast with tf 1.3

it looks there is no big change for beam search decoder op between 1.3 and 1.4. googling the problem got no luck either.

can you please take a look?
we could provide the pb file and a small inference script if needed. since pb file size is big, we do not upload it at this point
thanks.

**Describe the expected behavior**

**Code to reproduce the issue**
Provide a reproducible test case that is the bare minimum necessary to generate the problem.

**Other info / logs**
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.

**Comment by tianq01** [comment]
more info:
the test for tf 1.12, 1.9, 1.8, 1.5, 1.4 are all in docker env
the test for tf 1.3 covers docker env and host env

**Comment by tianq01** [comment]
@ebrevdo could you help take a look?
thanks

**Comment by ebrevdo** [comment]
The only PR i can think that may have affected you is [this one](https://github.com/tensorflow/tensorflow/commit/4503f464628d4ba6f01e0e5b2aa9ff829763982b).  You could try building TF before and after this PR and see if it is in fact causing the regression.

**Comment by tianq01** [comment]
@ebrevdo thanks for response!
the PR was introduced in tf 1.5? but the problem happens on tf 1.4 already...

**Comment by jvishnuvardhan** [comment]
@tianq01 Could you check whether the bug persists with TF2.0? Could you also check whether the issue is with docker env only? Thanks!

**Comment by tianq01** [comment]
thanks @jvishnuvardhan , the issue is not related to docker env.
tf 2.0 might be heavy to try? any other suggestions that could help investigate and narrow down the problem?

**Comment by tianq01** [comment]
how are you doing @ebrevdo ?
any update? many thanks

**Comment by zihaolucky** [comment]
@tianq01 what's your size of the output space? and beam_width?

**Comment by rmothukuru** [comment]
> thanks @jvishnuvardhan , the issue is not related to docker env.
> tf 2.0 might be heavy to try? any other suggestions that could help investigate and narrow down the problem?
> 
> @ebrevdo @jvishnuvardhan FYI timeline gave the same conclusion: decoder op is the bottleneck

@tianq01,
May be you can give a try with **`Tensorflow Version 1.15.2`**.
